### ISSUE TITLE
DSN name in prompt

### ISSUE THREAD

**Issue by tsroten** [dbcli/mycli#582]
Similar to how we can currently put the login path host name in the prompt, we should be able to do that with the DSN settings.

**Comment by j-bennet** [comment]
@Man-Jain Certainly, feel free to submit a PR. Good luck!

**Comment by Man-Jain** [comment]
@j-bennet Can you please guide me on how can I begin.

**Comment by j-bennet** [comment]
We had a few PRs in the past that deal with changing prompt, you can use them as examples:

https://github.com/dbcli/mycli/pull/275
https://github.com/dbcli/mycli/pull/458
https://github.com/dbcli/mycli/pull/323

What you want to do will be very similar to https://github.com/dbcli/mycli/pull/275: display something else instead of host name instead on `\\h` placeholder, based on a number of conditions (in this case, if dsn name is provided, display that instead).

**Comment by Man-Jain** [comment]
@j-bennet Thanks I will start working on this. :)

**Comment by Man-Jain** [comment]
@j-bennet Should i implement this as a new promt or as you said earlier replace by host. Cause I am not able to understand that if the dsn will replace the hoat and the default prompt is provided then the other fields will still be there. Please correct me if I am I wrong.

**Comment by j-bennet** [comment]
@Man-Jain `pgcli` has a special placeholder for `\dsn_alias` in prompt, you could do the same:

https://github.com/dbcli/pgcli/blob/5834259535e1cf5a75204305b0533ac5559ff344/pgcli/main.py#L911

The other fields would still be there, true, but it's not necessarily a problem.

**Comment by j-bennet** [comment]
@Man-Jain Not really, since you would display an alias, not the dsn string itself:
```
example_dsn = postgresql://[user[:password]@][netloc][:port][/dbname]
^^^           ^^^
show this     not this
```

**Comment by Man-Jain** [comment]
Okay, Got it

**Comment by rolandwalker** [comment]
This feature was added!  The abbreviation is `\A`.
